src/Workspaces/Core/Portable/NamingStyles/Serialization/SymbolSpecification.cs

```diff
@@ -228,9 +229,35 @@ internal static SymbolKindOrTypeKind AddTypeKindFromXElement(XElement typeKindEl
             {
                 return new SymbolKindOrTypeKind((TypeKind)Enum.Parse(typeof(TypeKind), typeKindElement.Value));
             }
+
+            public override bool Equals(object obj)
+            {
+                return Equals(obj as SymbolKindOrTypeKind);
+            }
+
+            public bool Equals(SymbolKindOrTypeKind other)
+            {
+                return other != null &&
+                       this.SymbolKind == other.SymbolKind &&
+                       this.TypeKind == other.TypeKind;
+            }
+
+            public override int GetHashCode()
```

Comment:
Why not just `Hash.Combine(SymbolKind.GetHashCode(), TypeKind.GetHashCode())`. Since nullable is a struct you can call the GetHashCode without concern for it's underlying nullability.